modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/permission/CommerceCatalogPermissionImpl.java

```diff
@@ -145,7 +147,10 @@ private boolean _contains(
 				if (commerceCatalog.getAccountEntryId() ==
 						accountEntry.getAccountEntryId()) {
 
-					return true;
+					return _roleLocalService.hasUserRole(
+						permissionChecker.getUserId(),
+						permissionChecker.getCompanyId(),
+						AccountRoleConstants.ROLE_NAME_SUPPLIER, true);
```

Comment:
@luca-pellizzon I added in supplier role check to align with @brianikim's permission logic.